internal/api/integrations/k8s/k8s_test.go

```diff
@@ -0,0 +1,55 @@
+package k8s
+
+import (
+	"encoding/json"
+	"testing"
+
+	shipitv1beta1 "ship-it-operator/api/v1beta1"
+	"ship-it/internal/api/models"
+
+	"github.com/stretchr/testify/assert"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+func TestDockerArtifacts(t *testing.T) {
```

Comment:
This is a test for a private function in package. What we should be doing instead is testing the public function exposed by this package `ListAll` and asserting that the docker artifacts get returned.

Reply:
Testing `ListAll` is going to be slightly hard because we'll need to mock the List method on the k8s client.

Could use this: https://godoc.org/sigs.k8s.io/controller-runtime/pkg/client/fake

Reply:
The `K8sClient` is difficult to unit test because `New()` creates a controller-runtime client from a `rest.InClusterConfig`, depending on a real cluster environment.

I could try to use the [fake client](https://godoc.org/sigs.k8s.io/controller-runtime/pkg/client/fake), but then I'm assigning to the private `K8sClient.client` field, which is related to the same issue you're calling out here.

So before I go forward with the fake client, are we okay with that approach?